src/features/payment/components/PaymentChoice/PaymentChoice.tsx

```diff
@@ -1,92 +1,147 @@
-    enabled: enabled.slice().sort(sortMethods),
-    disabled: disabled.slice().sort(sortMethods),
-    duplicatedMethods,
-  };
 };
 
 export function PaymentChoice(props: {
   amount: number;
   paymentInstruments: Array<PaymentInstrumentsType>;
   loading?: boolean;
 }) {
-  const handleClickOnMethod = React.useCallback(
-    (paymentTypeCode: PaymentCodeType, paymentMethodId: string) => {
-      const route: string = PaymentMethodRoutes[paymentTypeCode]?.route;
-      setSessionItem(SessionItems.paymentMethod, {
-        paymentMethodId,
-        paymentTypeCode,
-      });
+  const ref = React.useRef<ReCAPTCHA>(null);
+  const [loading, setLoading] = React.useState(true);
+
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+
+  React.useEffect(() => {
+    if (ref.current) {
+      setLoading(false);
+    }
+  }, [ref.current]);
+
+  const onError = () => {
+    setLoading(false);
+    ref.current?.reset();
+  };
+
+  const getFees = (onSuccess: () => void) =>
+    calculateFees({
+      paymentId:
+        (
+          getSessionItem(SessionItems.paymentMethod) as
+            | PaymentMethod
+            | undefined
+        )?.paymentMethodId || "",
+      onError,
+      onResponsePsp: (resp) => {
+        pipe(
+          resp,
+          CalculateFeeResponse.decode,
+          O.fromEither,
+          O.chain((resp) => O.fromNullable(resp.belowThreshold)),
+          O.fold(
+            () => onError(),
+            (value) => {
+              dispatch(setThreshold({ belowThreshold: value }));
+              const firstPsp = pipe(
+                resp?.bundles,
+                O.fromNullable,
+                O.chain((sortedArray) => O.fromNullable(sortedArray[0])),
+                O.map((a) => a as Bundle),
+                O.getOrElseW(() => ({}))
+              );
+
+              setSessionItem(SessionItems.pspSelected, firstPsp);
+              onSuccess();
+            }
+          )
+        );
+      },
+    });
+
+  const transaction = async (
+    recaptchaRef: ReCAPTCHA,
+    onSuccess: () => void
+  ) => {
+    setLoading(true);
+    const token = await callRecaptcha(recaptchaRef, true);
+    await activatePayment({
+      token,
+      onResponseActivate: async () => {
+        await getFees(onSuccess);
+      },
+      onErrorActivate: onError,
+    });
+  };
+
+  const onSuccess = (paymentTypeCode: PaymentCodeType) => {
+    const route: string = PaymentMethodRoutes[paymentTypeCode]?.route;
+    setLoading(false);
+    navigate(`/${route}`);
+  };
+
+  const handleClickOnMethod = async (method: PaymentInstrumentsType) => {
+    if (!loading) {
+      const { paymentTypeCode, id: paymentMethodId } = method;
       mixpanel.track(PAYMENT_METHODS_CHOICE.value, {
         EVENT_ID: paymentTypeCode,
       });
 
-      window.location.assign(`/${route}`);
-    },
-    []
-  );
+      setSessionItem(SessionItems.paymentMethodInfo, {
+        title: method.name,
+        body: method.description,
```

Comment:
Leave `body` empty and use the `description` value as title

Reply:
done